server/src/engine/flix.ts

```diff
@@ -76,7 +76,12 @@ export function getExtensionVersion() {
 }
 
 export function getProjectRootUri() {
-  return _.first(startEngineInput.workspaceFolders)
+  const path = _.first(startEngineInput.workspaceFolders)!
+
+  // This is the format that that VSCode forwards the URI in
+  const escapedPath = path.replace(':', '%3A')
```

Comment:
Can this be right? Does the internet suggest anything better?

Reply:
It's the only thing that works, and I have never managed to find any useful documentation about the server side of VSCode, but I'll welcome any proposed search terms.

Reply:
Here is a page where they are seemingly encouraging you to implement your language server in HTML: https://code.visualstudio.com/api/language-extensions/overview

Reply:
decodeUri?

https://github.com/Microsoft/vscode-uri